### ISSUE TITLE
CMake doesn't detect MIPS variant properly

### ISSUE THREAD

**Issue by neheb** [libjpeg-turbo/libjpeg-turbo#248]
From my reading of the SIMD code in here, it's only valid for processors with DSPr2. It seems MIPS 74kc more specifically. It's not looking for the 74kc. 

It also thinks that mipsel != mips. I don't know if the assembly is written with big endian in mind.

**Comment by dcommander** [comment]
I have no way to test non-DSPr2 MIPS systems, and actually, the only way I have in which to test DSPr2-enabled MIPS systems is via a Docker/QEMU container, which is really slow.  Absolutely nothing will happen on this front unless someone provides access to a test platform.

**Comment by neheb** [comment]
This was discovered through adding support to OpenWrt. Here's a sample log: https://travis-ci.org/openwrt/packages/builds/395313734?utm_source=github_status&utm_medium=notification

**Comment by dcommander** [comment]
Sample log does me no good.  I need to add a mechanism for testing whether the compiler supports DSPr2, and that requires access to both a positive (DSPr2) and negative (non-DSPr2) build environment.

**Comment by dcommander** [comment]
I've tried various values of `-march`, attempting to find one that reproduces the failure, but no success.  As I said, I need a build environment that lacks support for DSPr2 instructions, in order to test the proposed compile-time detection mechanism.  Otherwise this will not get fixed.  Obviously you have such a cross-compilation environment-- the one being used by your Travis builds.  Please point me to the same cross-compiler you are using.

**Comment by neheb** [comment]
Sure. It's a standard OpenWrt installation. To set it up, 

git clone https://git.openwrt.org/openwrt/openwrt.git
cd openwrt
./scripts feeds update -a
cd feeds/packages
git pr 6331
cd ~/OpenWrt
./scripts feeds install -a
make menuconfig

Select libjpeg-turbo.

make V=s

Will take a long time for the toolchain to build.

**Comment by dcommander** [comment]
Is there not a prebuilt toolchain?  I tried [this one](https://github.com/MatteoRagni/ArduinoYun-x86_64-OpenWRT-mips-linux-toolchain), which-- from your Travis log-- appears to be what you are using, but I can't reproduce the failure with that toolchain.  Thus, I have no confidence that the problem will be reproducible by building the toolchain from scratch.

**Comment by neheb** [comment]
That toolchain is alien. The OpenWrt toolchain uses GCC 7.3.0. I can reproduce the issue locally.

It doesn't matter either way. If GCC is able to compile it, libjpeg-turbo will probably crash since DSPr2 instructions mainly exist and are coded for MIPS 74k CPUs.

I also just realized my instructions won't work. I patched out the SIMD MIPS stuff. Need to remove openwrt/feeds/packages/libs/libjpeg-turbo/patches/mips-patch.patch
